### PR TITLE
Connect Vehicles and Maintenance Events

### LINKED ISSUE #141: FEATURE - Connect `Vehicle`s and `MaintenanceEvent`s
## Describe the feature you want the app to have that it currently doesn't
* The `Vehicle` object needs to correspond to a `MaintenanceEvent`
* When creating a new `MaintenanceEvent` a new row needs to be added so that a user can select which `Vehicle` this event belongs to.
* I want to make a [menu like this](https://www.hackingwithswift.com/quick-start/swiftui/how-to-let-users-pick-options-from-a-menu)

## Describe alternatives you've considered
* There's many ways to implement this, at the moment the easiest is probably just fetching the vehicles inside of the `AddMaintenanceView`, in a function. Because otherwise it would require rewriting too much of the app at this point

## Additional context
* Also update the `EditEventDetailView` with the new property
* Once this is done, the widget tasks can be started (which are all separate for each widget size)

### PR DESCRIPTION
**Pull request by hanhqvu**
# What it Does
* Closes #141
* This PR will add a Picker (pop-up) menu to `AddMaintenanceView `& `EditMaintenanceView`

# How I Tested
* I added sample vehicles data to show in the preview

# Notes
* There is no validation upon adding/editing to ensure a vehicle has been selected

# Screenshot
Screenshot 2023-10-12 at 23 36 14

### CODE REVIEW

File: Basic-Car-Maintenance/Shared/Dashboard/Views/AddMaintenanceView.swift [hunk 1]
```diff
@@ -31,6 +33,21 @@ struct AddMaintenanceView: View {
...
+                    } label: {
+                        Text("Select a vehicle",
+                             comment: "Maintenance event vehicle picker label")
+                    }
+                    .pickerStyle(.menu)
+                } header: {
+                    Text("Vehicle",
+                         comment: "Maintenance event vehicle picker header")
+                }
+
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/AddMaintenanceView.swift]: This isn't keeping track of the vehicle that's selected. See video

https://github.com/mikaelacaron/Basic-Car-Maintenance/assets/22946902/c3da09b6-762f-409b-8a5b-dc43765b2d2b

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
It's almost there! In the `AddMaintenanceView` and the `EditEventDetailView` it's not keeping track when you select a vehicle in the list, the row isn't changing.

Also can you show which vehicle is selected, in the `DashboardView`

**Comment by hanhqvu** [comment]
> It's almost there! In the `AddMaintenanceView` and the `EditEventDetailView` it's not keeping track when you select a vehicle in the list, the row isn't changing.
> 
> Also can you show which vehicle is selected, in the `DashboardView`

image

Regarding showing which vehicle is selected, do you mean like in the picture?

**Comment by mikaelacaron** [comment]
> Regarding showing which vehicle is selected, do you mean like in the picture?

Yes just show the name, and it's fine to just add it into the `VStack` that's already there (it can be a separate issue for someone to style it better)

**Comment by mikaelacaron** [comment]
Also be sure to merge the `dev` branch into your branch (and resolve the merge conflicts)

**Comment by hanhqvu** [comment]
@mikaelacaron I added the fix for the picker menu & added the vehicle info to the Dashboard view

**Review by mikaelacaron — APPROVED** [review]
I added two changes to `EditEventDetailView` and `AddMaintenanceView` where to use an optional rather than an empty `selectedVehicle` object

**Comment by hanhqvu** [comment]
@mikaelacaron Thank you for the changes. I added #174 so someone can style `DashboardView` better!

**Pull request merged**